sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/executeImmediate.scala

```diff
@@ -63,6 +63,7 @@ class SubstituteExecuteImmediate(val catalogManager: CatalogManager)
         getVariableReference(u, u.nameParts)
       case a: Alias =>
         Alias(resolveVariable(a.child), a.name)()
+      case p: Parameter => p
```

Comment:
can we add a comment to explain how can it happen?

Reply:
I updated the comment.